src/core/sagas/products.ts

```diff
@@ -11,24 +13,53 @@ import { Tasks as productDetailsTasks } from './product-details';
 export namespace Tasks {
   export function* fetchProducts(flux: FluxCapacitor, action: Actions.FetchProducts) {
     try {
-      const request = yield effects.select(Requests.search, flux.config);
-      const res = yield effects.call([flux.clients.bridge, flux.clients.bridge.search], request);
-
-      if (res.redirect) {
-        yield effects.put(flux.actions.receiveRedirect(res.redirect));
+      const [products, navigations]: [Results, Store.Recommendations.Navigation[]] = yield effects.all([
+        effects.call(fetchProductsRequest, flux, action),
+        effects.call(fetchNavigations, flux, action)
+      ]);
+      if (products.redirect) {
+        yield effects.put(flux.actions.receiveRedirect(products.redirect));
       }
-
-      if (flux.config.search.redirectSingleResult && res.totalRecordCount === 1) {
-        yield effects.call(productDetailsTasks.receiveDetailsProduct, flux, res.records[0]);
+      if (flux.config.search.redirectSingleResult && products.totalRecordCount === 1) {
+        yield effects.call(productDetailsTasks.receiveDetailsProduct, flux, products.records[0]);
       } else {
-        flux.emit(Events.BEACON_SEARCH, res.id);
-        yield effects.put(<any>flux.actions.receiveProducts(res));
+        flux.emit(Events.BEACON_SEARCH, (<any>products).id);
+        if (navigations && !(navigations instanceof Error)) {
+          products.availableNavigation = RecommendationsAdapter.sortAndPinNavigations(products.availableNavigation,
```

Comment:
if you really want to break parameters onto multiple lines, the preferred format is
```js
products.availableNavigation = RecommendationsAdapter.sortAndPinNavigations(
  products.availableNavigation,
  navigations,
  flux.config
);
```